src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj

```diff
@@ -1,145 +1,182 @@
+            project, when msbuild evaluation runs the published server artifacts won't exist and so the content will be empty.
+  -->
+  <Target Name="PublishAllRids" DependsOnTargets="Publish">
+    <Message Text="Publishing builds for each runtime identifier..." Importance="High" />
 
     <ItemGroup>
-        <EmbeddedResource Update="LanguageServerResources.resx" GenerateSource="true" />
+      <!-- Transform RuntimeIdentifiers property to item -->
+      <RuntimeIdentifierForPublish Include="$(RuntimeIdentifiers)" />
+
+      <!-- Transform RuntimeIdentifierForPublish items to project items to pass to MSBuild task -->
+      <ProjectToPublish Include="@(RuntimeIdentifierForPublish->'$(MSBuildProjectFullPath)')">
+        <AdditionalProperties>RuntimeIdentifier=%(RuntimeIdentifierForPublish.Identity)</AdditionalProperties>
+      </ProjectToPublish>
     </ItemGroup>
 
-    <Target Name="WriteRoslynVersion" AfterTargets="Build">
-        <WriteLinesToFile File="@(RoslynVersionFile)" Lines="$(MicrosoftCodeAnalysisVersion)" Overwrite="true" />
-    </Target>
+    <!--
+        We have to run restore first with the PublishReadyToRun flag set to true to ensure that the correct crossgen packages get restored.
+        See https://github.com/dotnet/sdk/issues/20701
+    -->
+    <MSBuild Projects="$(MSBuildProjectFullPath)" Targets="Restore" Properties="RestoreWithR2R=true" />
 
-    <Target Name="PublishRoslynVersion" AfterTargets="Publish">
-        <Copy SourceFiles="@(RoslynVersionFile)" DestinationFolder="$(PublishDir)" />
-    </Target>
+    <MSBuild Projects="@(ProjectToPublish)" Targets="Publish" BuildInParallel="true" />
 
-    <!--
-        This target runs publishing for all of the RIDs that we've specified in the project to create platform specific
-        executables.  It then updates the Content in the project to include the executables we built in the NuGet package.
+    <!-- We have to add the content files in a target (and not just defined in a normal item group) since they are produced by the prior step and won't exist at evaluation time. -->
+    <Message Text="Adding content files for language server artifacts..." Importance="High" />
+    <ItemGroup>
+      <Content Include="$(ArtifactsDir)/LanguageServer/$(Configuration)/$(TargetFramework)/**/*.*">
+        <Pack>true</Pack>
+        <PackagePath>content\LanguageServer</PackagePath>
+        <CopyToOutput>false</CopyToOutput>
+        <BuildAction>None</BuildAction>
+      </Content>
+    </ItemGroup>
+  </Target>
 
-        This is also partially a workaround for lack of support for publishing multiple RIDs directly in a csproj.  See https://github.com/dotnet/sdk/issues/9363
+  <!--
+    Deploy a net472 version of the BuildHost process.
+  -->
+  <Target Name="DeployNetFrameworkBuildHost" AfterTargets="ResolveProjectReferences" Condition="'$(DesignTimeBuild)' != 'true'">
```

Comment:
once this becomes a part of the msbuild workspace, does this all move into that project?

Reply:
Yes.